azure-pipelines.yml

```diff
@@ -43,127 +43,45 @@ stages:
     dependsOn: BUILD
     displayName: Tests - Functional
     jobs:
-      - job: testfunc
-        pool:
-          vmImage: 'windows-latest'
-        steps:
-
-          - task: Gradle@2
-            inputs:
-              gradleWrapperFile: 'gradlew'
-              workingDirectory: '$(System.DefaultWorkingDirectory)'
-              tasks: clean functional -DTEST_SUBSCRIPTION_KEY=$(hmi-apim-sub-key) -DTEST_URL=https://hmi-apim-svc-sbox.azure-api.net -DTEST_HOST=hmi-apim-svc-sbox.azure-api.net
-              publishJUnitResults: true
-              testResultsFiles: '**/TEST-*.xml'
-              javaHomeOption: 'JDKVersion'
-              jdkVersionOption: '1.11'
-              sonarQubeRunAnalysis: false
-
-          - task: AzureFileCopy@4
-            displayName: 'Publish gradle output'
-            inputs:
-              SourcePath: '$(System.DefaultWorkingDirectory)/src/functionalTest/resources/'
-              azureSubscription: $(subscription)
-              Destination: AzureBlob
-              storage: $(devopsStorageAccount)
-              ContainerName: '$web'
-            condition: succeededOrFailed()
+    - job: testfunc
+      variables:
+        hmi-apim-sub-key: $[stageDependencies.BUILD.Terraform.outputs['exportKeyVaultSecret.hmi-apim-sub-key']]
+        hmi-apim-host: hmi-apim-svc-sbox.azure-api.net
+      pool:
+        vmImage: 'windows-latest'
+      steps:
+      - template: pipeline/steps/run-gradlew.yaml
+        parameters:
+          runCommand: clean functional -DTEST_SUBSCRIPTION_KEY=$(hmi-apim-sub-key) -DTEST_URL=https://hmi-apim-svc-sbox.azure-api.net -DTEST_HOST=$(hmi-apim-host)
 
-          - task: AzureFileCopy@4
-            displayName: 'Publish serenity output'
-            inputs:
-              SourcePath: '$(System.DefaultWorkingDirectory)/target/site/serenity'
-              azureSubscription: $(subscription)
-              Destination: AzureBlob
-              storage: $(devopsStorageAccount)
-              ContainerName: '$web'
-            condition: succeededOrFailed()
 
-  - stage: TESTS4
+  - stage: TESTS2
     dependsOn: BUILD
     displayName: Tests - Smoke
-    variables:
-      hmi-apim-svc-sub-key: stageDependencies.BUILD.Terraform.outputs['exportKeyVaultSecret.variableName']
     jobs:
     - job: testsmoke
+      variables:
+        hmi-apim-sub-key: $[stageDependencies.BUILD.Terraform.outputs['exportKeyVaultSecret.hmi-apim-sub-key']]
+        hmi-apim-host: hmi-apim-svc-sbox.azure-api.net
```

Comment:
Does the above variable get overwritten when the build runs for test environment?

Reply:
Yes, this is for SBOX only as TEST pipeline is currently a WiP.
Ultimately $HOST will be read from terraform output after build stage, for now it's as was